editor/src/components/inspector/common/control-status.ts

```diff
@@ -39,53 +33,6 @@ export interface ControlStyles {
-  SetSegmentSelectorColor: colorTheme.inspectorSetBackgroundColor.value,
-  SetSegmentTrackColor: colorTheme.inspectorSetSegmentTrackColor.value,
-  UnsetMainColor: colorTheme.inspectorUnsetMainColor.value,
-  UnsetSecondaryColor: colorTheme.inspectorUnsetSecondaryColor.value,
-  UnsetBorderColor: colorTheme.inspectorUnsetBorderColor.value,
-  UnsetBorderHoverFocusColor: colorTheme.inspectorSetBorderColor.value,
-  UnsetBackgroundColor: colorTheme.inspectorUnsetBackgroundColor.value,
-  UnsetSegmentSelectorColor: colorTheme.inspectorUnsetSegmentSelectorColor.value,
-  UnsetSegmentTrackColor: colorTheme.inspectorUnsetSegmentTrackColor.value,
-  DisabledMainColor: colorTheme.inspectorDisabledMainColor.value,
-  DisabledSecondaryColor: colorTheme.inspectorDisabledSecondaryColor.value,
-  DisabledBackgroundColor: colorTheme.inspectorDisabledBackgroundColor.value,
-  DisabledSegmentSelectorColor: colorTheme.inspectorDisabledSegmentSelectorColor.value,
-  DisabledSegmentTrackColor: colorTheme.inspectorDisabledSegmentTrackColor.value,
-  DisabledBorderColor: colorTheme.inspectorDisabledBorderColor.value,
-  UneditableMainColor: colorTheme.inspectorUneditableMainColor.value,
-  UneditableSecondaryColor: colorTheme.inspectorUneditableSecondaryColor.value,
-  UneditableBackgroundColor: colorTheme.inspectorUneditableBackgroundColor.value,
-  UneditableBorderColor: colorTheme.inspectorUneditableBorderColor.value,
-  ControlledMainColor: colorTheme.inspectorControlledMainColor.value,
-  ControlledSecondaryColor: colorTheme.inspectorControlledMainColor.value,
-  ControlledBorderColor: colorTheme.inspectorControlledBorderColor.value,
-  ControlledBackgroundColor: colorTheme.inspectorControlledBackgroundColor.value,
-  ControlledSegmentSelectorColor: colorTheme.inspectorControlledSegmentSelectorColor.value,
-  ControlledSegmentTrackColor: colorTheme.inspectorControlledSegmentTrackColor.value,
-  DetectedMainColor: colorTheme.inspectorDetectedMainColor.value,
-  DetectedSecondaryColor: colorTheme.inspectorDetectedMainColor.value,
-  DetectedBorderColor: colorTheme.inspectorDetectedBorderColor.value,
-  DetectedBackgroundColor: colorTheme.inspectorDetectedBackgroundColor.value,
-  DetectedSegmentSelectorColor: colorTheme.inspectorDetectedSegmentSelectorColor.value,
-  DetectedSegmentTrackColor: colorTheme.inspectorDetectedSegmentTrackColor.value,
-  DetectedFromCssMainColor: colorTheme.inspectorDetectedFromCssMainColor.value,
-  DetectedFromCssSecondaryColor: colorTheme.inspectorDetectedFromCssMainColor.value,
-  OffMainColor: colorTheme.inspectorOffMainColor.value,
-  OffSecondaryColor: colorTheme.inspectorOffSecondaryColor.value,
-  OffBackgroundColor: colorTheme.inspectorOffBackgroundColor.value,
-  OffSegmentSelectorColor: colorTheme.inspectorOffSegmentSelectorColor.value,
-  OffSegmentTrackColor: colorTheme.inspectorOffSegmentTrackColor.value,
-  OffBorderColor: colorTheme.inspectorOffBorderColor.value,
-}
-
 export type ControlStatus =
```

Comment:
two observations:

1. this function tries to reduce repetition by only overwriting a few values, but this actually makes it harder to parse, vs every case just returning a full style object – we have the theme anyways, if you want to mass update a color
2. this function was made years ago, we later introduced the `PropertyStatus` type which is an object with fields like controlled, set, overwritable, detected, fromCssStyleSheet, trivialDefault etc – there's no need to condense these to harder-to-understand statuses like `multiselect-identical-unset`. We could write a styling function that works much more directly (ie `if trivialDefault, set to light grey`)

Reply:
I was thinking similar while reading this, it would be easier to have full style objects where you can see all properties in one place

Reply:
> "every function returning a full style object"  / "see all properties in one place"

Admittedly the previous code _was_ closer to this, at least in step 1. One of the downsides was that - more so now than before - most control statuses only change a few values, many just two or three, and returning everything made it hard to see what actually is different.

The other downside - and I did try returning style _objects_ experimentally - is that you basically need to reconstruct your components inside the returned object: the controls often apply the colours in specific places, and almost never do those live on the outer, containing, element. Even on the humble "input"!